packages/medusa/src/api/utils/middlewares/products/filter-by-valid-sales-channels.ts

```diff
@@ -1,77 +1,49 @@
 // - If a publishable API key is passed, we take the sales channels attached to it and filter them down based on the query params
 // - If a sales channel id is passed through query params, we use that
 // - If not, we use the default sales channel for the store
 export function filterByValidSalesChannels() {
-  return async (req: AuthenticatedMedusaRequest, _, next: NextFunction) => {
-    const publishableApiKey = req.get("x-publishable-api-key")
-    const salesChannelIds = req.filterableFields.sales_channel_id as
-      | string[]
-      | undefined
-
-    if (publishableApiKey) {
-      const apiKey = await refetchEntity(
-        "api_key",
-        { token: publishableApiKey },
-        req.scope,
-        ["id", "sales_channels_link.sales_channel_id"]
+  return async (req: MedusaStoreRequest, _, next: NextFunction) => {
+    const idsFromRequest = req.filterableFields.sales_channel_id
+    const { sales_channel_ids: idsFromPublishableKey = [] } =
+      req.publishable_key_context
+
+    // If all sales channel ids are not in the publishable key, we throw an error
+    if (Array.isArray(idsFromRequest) && idsFromRequest.length) {
+      const uniqueInParams = arrayDifference(
+        idsFromRequest,
+        idsFromPublishableKey
       )
 
-      if (!apiKey) {
+      if (uniqueInParams.length) {
         return next(
           new MedusaError(
             MedusaError.Types.INVALID_DATA,
-            `Publishable API key not found`
+            `Requested sales channel is not part of the publishable key`
           )
         )
       }
-      let result = apiKey.sales_channels_link.map(
-        (link) => link.sales_channel_id
-      )
 
-      if (salesChannelIds?.length) {
-        // If all sales channel ids are not in the publishable key, we throw an error
-        const uniqueInParams = arrayDifference(salesChannelIds, result)
-        if (uniqueInParams.length) {
-          return next(
-            new MedusaError(
-              MedusaError.Types.INVALID_DATA,
-              `Requested sales channel is not part of the publishable key mappings`
-            )
-          )
-        }
-        result = salesChannelIds
-      }
+      req.filterableFields.sales_channel_id = idsFromRequest
 
-      req.filterableFields.sales_channel_id = result
       return next()
     }
 
-    if (salesChannelIds?.length) {
-      req.filterableFields.sales_channel_id = salesChannelIds
+    if (idsFromPublishableKey?.length) {
+      req.filterableFields.sales_channel_id = idsFromPublishableKey
+
       return next()
     }
 
-    const store = await refetchEntity("stores", {}, req.scope, [
-      "default_sales_channel_id",
-    ])
-
-    if (!store) {
-      return next(
-        new MedusaError(MedusaError.Types.INVALID_DATA, `Store not found`)
+    return next(
+      new MedusaError(
+        MedusaError.Types.INVALID_DATA,
+        `Publishable key needs to have a sales channel configured`
```

Comment:
question: why don't we use the default sales channel here anymore?

Reply:
This is an open question. Now that we require publishable keys on all store endpoints, should we push the developers to setup their publishable keys more explicitly than falling back to the default sales channel? wdyt?

Reply:
I think that fallback to the default is more convenient and that's why we have the default channel but no strong opinion there.